drs_filer/ga4gh/drs/endpoints/registerNewObject.py

```diff
@@ -0,0 +1,13 @@
+import logging
+from foca.errors.exceptions import handle_problem
+
+logger = logging.getLogger(__name__)
+
+
+def registerNewObject(db_collection, jsonData):
+    try:
```

Comment:
I suppose you should also coin somewhere a DRS identifier of the form outlined in the spec. I'm not sure how it should look like, but I think it should start with something like `drs://`. Please carefully read the specs

Reply:
@uniqueg Do you mean, I should add a validator to check if the `self_uri` should start with `drs://` as mentioned here https://github.com/ga4gh/data-repository-service-schemas/blob/master/openapi/data_repository_service.swagger.yaml?

Reply:
Yes, but this is slightly complex because:
- this is _not_ something that the user or admin should ever fill out in the `POST` / `PUT` endpoints
- this should _always_ be generated by the DRS itself, according to the rules described in the spec
- the implementation should be so generic that it will still work on different machines
- note that due to the host name, this is essentially a [UUID](https://en.wikipedia.org/wiki/Universally_unique_identifier); but it won't be if the host is an IP address - so what to do when there's no domain registered (yet) for the service? I don't know...

But these considerations I think you can put in a separate issue and deal with it later. But for sure this means that we will need a slightly different model for `POST` and `PUT` endpoints, I'm afraid. Same goes for the `id` field - it doesn't make sense to put that manually when creating an object. I need to think this through some more...